### PR TITLE
Adds the QuintoAndar case study

### LINKED ISSUE #6271: [Vitals][Runtime] Quinto Andar Core Web Vitals case study [1/1/2022]
* [Draft](https://docs.google.com/document/d/1Ct3y_ECv_ctIM3PhXAeRVMyPQJq4CxDjoS1niqSaVSw/edit)

### PR DESCRIPTION
**Pull request by malchata**
Fixes #6271 

Changes proposed in this pull request:
- Adds the QuintoAndar case study to the blog.

cc: @danielasy. There are some TODOs in the review. I'll tag you on them. It's all post meta stuff.

### CODE REVIEW

File: src/site/content/en/blog/quintoandar/index.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
+---
+title: How QuintoAndar increased conversion rates and pages per session by improving page performances
+subhead: A project focused on optimizing Core Web Vitals and migrating to Next.js resulted in a 5% increase in conversion rates and a 87% increase in pages per session.
+authors:
+  - jlwagner
```
Comment (malchata) [inline src/site/content/en/blog/quintoandar/index.md:5]: TODO: We need the names of authors (it wasn't evident from the doc), and a 384x384 headshot for each of them, as well as their title for their author data.

cc: @danielasy

File: src/site/content/en/blog/quintoandar/index.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+date: 2021-12-06
+hero: image/vgdbNJBYHma2o62ZqYmcnkq3j0o1/PHqBoMdg3WgcQWWIIkWj.jpg
```
Comment (malchata) [inline src/site/content/en/blog/quintoandar/index.md:7]: TODO: We need a 3200x960 image for the hero. The hero image in the doc wasn't sufficient in terms of dimensions.

cc: @danielasy

File: src/site/content/en/blog/quintoandar/index.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+alt: A picture of a city skyline on a clear, sunny day. At left is a blue box with the QuintoAndar logo set within it in white.
+description: |
+  TODO
```
Comment (malchata) [inline src/site/content/en/blog/quintoandar/index.md:10]:
```suggestion
  A project focused on optimizing Core Web Vitals and migrating to Next.js resulted in a 5% increase in conversion rates and a 87% increase in pages per session.
```

Comment (malchata) [reply src/site/content/en/blog/quintoandar/index.md:10]: TODO: Review this description. At current, it's just the same as the headline. If that's fine, I can just commit the above suggestion.

cc: @danielasy

File: src/site/content/en/blog/quintoandar/index.md [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+As for the impact in business metrics, we analyzed conversion rates for transactions like scheduling a tour and applying to rent or buy an estate. While improvements were still being rolled out, our team compared the conversion between the previous and new versions. In the same week, the group of pages with the new version showed a 5% conversion increase, while the other pages had a slight decrease in the same metric.
+
+<figure>
+  {% Img src="image/jL3OLOhcWUQDnR4XjewLBx4e3PC3/8duPNi2huaphIWCU7DBb.png", alt="Two line graphs side-by-side, each one comparing the conversion between the current and previous week. The left one is for the previous version of the page, showing the conversion curve for the current week is a bit below the one for the previous week. The right one is for the new version, and the conversion curve for the current week is a bit above the one for the previous week.", width="800", height="447" %}
+  <figcaption>
+    In the same week, the conversion for the new version increased, while the previous version had a small decrease.
+  </figcaption>
+</figure>
+
+## Conclusion
```
Comment (malchata) [inline src/site/content/en/blog/quintoandar/index.md:220]:
```suggestion
## Conclusion

```

**Comment by malchata** [comment]
Added DO_NOT_MERGE label to avoid an accidental merge until TODOs are addressed.

File: src/site/content/en/blog/quintoandar/index.md [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+
+### Optimizing JavaScript resources
+
+The first step was to [remove unused code](/remove-unused-code/). We looked at the [Webpack Bundle Analyzer](https://github.com/webpack-contrib/webpack-bundle-analyzer) reports, which shows the contents of each JS bundle, and carefully reviewed all third-party scripts. As a result, we were able to clean up some tracking libraries that were not used in this specific page.
+
+Our team went further and evaluated the performance cost of existing features. For instance, the "like" button required quite a lot of JS to work. However, in the condominium page, less than 0.5% of the users interacted with the button, which is available and used more frequently in other parts of our app. After a discussion involving both Engineering and Product, we decided to remove this feature.
+
+<figure>
+  {% Video src="video/jL3OLOhcWUQDnR4XjewLBx4e3PC3/2uEL2YxbuGGlmJDlvW9F.mp4", autoplay=true, muted=true, playsinline=true, loop=true, controls=true %}
```
Comment (danielasy) [inline src/site/content/en/blog/quintoandar/index.md:81]: Is there a way we can center the `<video>` within `<figure>`?

File: src/site/content/en/blog/quintoandar/index.md [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+  <figcaption>
+    An animation showing the “like” button feature. There is a card about an apartment available for rent. In the bottom right corner of the card, there is a grey heart-shaped button that turns blue when clicked.
+  </figcaption>
+</figure>
+
+Other JS optimizations were already in place, such as [static compression with Brotli](/reduce-network-payloads-using-text-compression/#static-compression), which was done at build time using [`BrotliWebpackPlugin`](https://github.com/mynameiswhm/brotli-webpack-plugin), and was also applied to other types of static resources. At first, we were relying on the compression provided by the CDN, and Brotli reduced JS size by 18% compared to gzip. But then, we switched to Brotli compression at build time, and were able to achieve a 24% reduction.
+
+### Optimizing image resources
+
+There is a hero image occupying most of the area above the fold in the mobile version. It also happens to be the [Largest Contentful Paint](//lcp/) (LCP) of the page.
```
Comment (danielasy) [inline src/site/content/en/blog/quintoandar/index.md:91]: Extra slash

```suggestion
There is a hero image occupying most of the area above the fold in the mobile version. It also happens to be the [Largest Contentful Paint](/lcp/) (LCP) of the page.
```

File: src/site/content/en/blog/quintoandar/index.md [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+
+<figure>
+  {% Img src="image/jL3OLOhcWUQDnR4XjewLBx4e3PC3/W7koRinBm3QXErIzwGP8.png", alt="The condominium page for Edifício Copan (São Paulo, Brazil). A photo taken from the ground level shows the curves of the building structure.", width="800", height="1422" %}
```
Comment (danielasy) [inline src/site/content/en/blog/quintoandar/index.md:94]: I think this image will look better scaled down:

```suggestion
  {% Img src="image/jL3OLOhcWUQDnR4XjewLBx4e3PC3/W7koRinBm3QXErIzwGP8.png", alt="The condominium page for Edifício Copan (São Paulo, Brazil). A photo taken from the ground level shows the curves of the building structure.", width="360", height="640" %}
```

Also, are we able to center this image?

File: src/site/content/en/blog/quintoandar/index.md [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+  <figcaption>
+    Caption: The hero image of a condominium page.
```
Comment (danielasy) [inline src/site/content/en/blog/quintoandar/index.md:96]:
```suggestion
    The hero image of a condominium page.
```

File: src/site/content/en/blog/quintoandar/index.md [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+
+### Reducing layout shift
+
+The condominium page had a few issues with [Cumulative Layout Shift](/cls/) (CLS). The elements responsible for the layout shifts were rendered only in the client — for instance, hydrating server-side markup with client-rendered components, or images without defined `width` and `height` attributes.
+
+To solve these problems, we set exact dimensions for these elements when possible, or estimated values with `min-height`. There are more options, such as using the [`aspect-ratio` CSS property](https://developer.mozilla.org/docs/Web/CSS/aspect-ratio). We also created placeholders to prevent dynamically rendered components from causing layout shifts.
+
+<figure>
+  {% Img src="image/jL3OLOhcWUQDnR4XjewLBx4e3PC3/VqFwUiJOtNmyNc14xljP.png", alt="An image showing an urban area in Google Maps with a red marker in the center.", width="397", height="174" %}
```
Comment (danielasy) [inline src/site/content/en/blog/quintoandar/index.md:119]: Same here about centering the image

**Review by danielasy — COMMENTED** [review]
Thanks a lot @malchata!

I have not received a response about the hero image yet, will send it via email as soon as I have it =)

File: src/site/content/en/blog/quintoandar/index.md [hunk 10]
```diff
@@ -0,0 +1,226 @@
+---
+title: How QuintoAndar increased conversion rates and pages per session by improving page performances
```
[the remark this answers is not in the document]
Comment (malchata) [reply src/site/content/en/blog/quintoandar/index.md:2]: Looks like a mistake to me, so good catch on this!

File: src/site/content/en/blog/quintoandar/index.md (hunk 5 above)
Comment (malchata) [reply src/site/content/en/blog/quintoandar/index.md:72]: Yep. Fixed this and all other containers.

**Comment by malchata** [comment]
@danielasy, when the deploy preview updates, please take a look and verify all is well. If so, we'll merge and deploy this ASAP!

File: src/site/content/en/blog/quintoandar/index.md [hunk 11]
```diff
@@ -0,0 +1,219 @@
...
+
+- They are an important source of [organic traffic](https://en.wikipedia.org/wiki/Organic_search_results), with a steadily increasing number of users coming from search engine results.
+- They have high conversion rates in the medium to long-term compared to other pages.
+
+However, there were challenges when it came to the performance and user experience in these pages:
+
+- Their performance as measured by [Core Web Vitals](/vitals/) was not optimized, and there were known issues regarding slow page loads, slow responsiveness to user input, and layout instability.
+- Their [bounce rates](https://en.wikipedia.org/wiki/Bounce_rate) were high, even if we expected them to be higher than in other parts of the app.
+- The [page experience update in Google Search](https://developers.google.com/search/blog/2020/11/timing-for-page-experience) — which, at that time, was not yet released — would include Core Web Vitals into the ranking algorithm, meaning page performance could affect how search results were going to be displayed.
```
Comment (heyawhite) [inline src/site/content/en/blog/quintoandar/index.md:45]:
```suggestion
- The [page experience update in Google Search](https://developers.google.com/search/blog/2020/11/timing-for-page-experience)&mdash;which, at that time, was not yet released&mdash;would include Core Web Vitals into the ranking algorithm, which meant page performance could affect how search results were going to be displayed.
```

File: src/site/content/en/blog/quintoandar/index.md [hunk 12]
```diff
@@ -0,0 +1,219 @@
...
+
+At the same time, we identified some developer experience opportunities that could unlock gains in other projects across the company:
+
+- Our server-side rendering logic — which renders all high-traffic pages, including condominium pages — was created in-house, and became too complex to maintain and onboard new hires.
```
Comment (heyawhite) [inline src/site/content/en/blog/quintoandar/index.md:49]:
```suggestion
- Our server-side rendering logic&mdash;which renders all high-traffic pages, including condominium pages&mdash;was created in-house, and became too complex to maintain and onboard new hires.
```

File: src/site/content/en/blog/quintoandar/index.md [hunk 13]
```diff
@@ -0,0 +1,219 @@
...
+
+A hard requirement was to ensure pages remained crawlable by search engines. Next.js meets this requirement by supporting server-side rendering out-of-the-box, and removes the need for a custom setup. The documentation makes it much easier to share knowledge on how to do tasks such as [data fetching](https://nextjs.org/docs/basic-features/data-fetching) on the server and onboard new developers. Server-side rendering is also known to [improve performance](https://developers.google.com/web/updates/2019/02/rendering-on-the-web#server-rendering) metrics such as [First Contentful Paint](/fcp/) (FCP).
+
+The framework provides other performance-friendly features such as automatic [code splitting](/reduce-javascript-payloads-with-code-splitting/) and [prefetching](/link-prefetch/). Even though the existing structure already provided such features, the additional work required from developers stalled their adoption — for example, code splitting at page or component-level had to be done manually.
```
Comment (heyawhite) [inline src/site/content/en/blog/quintoandar/index.md:63]:
```suggestion
The framework provides other performance-friendly features such as automatic [code splitting](/reduce-javascript-payloads-with-code-splitting/) and [prefetching](/link-prefetch/). Even though the existing structure already provided such features, the additional work required from developers stalled their adoption. For example, code splitting at page or component-level had to be done manually.
```

File: src/site/content/en/blog/quintoandar/index.md [hunk 14]
```diff
@@ -0,0 +1,219 @@
...
+
+```html
+<link rel="preload" href="/img/450x450/892847321-143.0038687080606IMG20180420WA0037.jpg" as="image">
+```
+
+The [Next.js built-in image component](https://nextjs.org/docs/basic-features/image-optimization) includes many of these optimizations such as responsive resizing and prioritized loading. During this project, we did not migrate the existing images to use this component, but we are planning to adopt it in new features.
+
+### Reducing layout shift
+
+The condominium page had a few issues with [Cumulative Layout Shift](/cls/) (CLS). The elements responsible for the layout shifts were rendered only in the client — for instance, hydrating server-side markup with client-rendered components, or images without defined `width` and `height` attributes.
```
Comment (heyawhite) [inline src/site/content/en/blog/quintoandar/index.md:105]:
```suggestion
The condominium page had a few issues with [Cumulative Layout Shift](/cls/) (CLS). The elements responsible for the layout shifts were rendered only in the client&mdash;for instance, hydrating server-side markup with client-rendered components, or images without defined `width` and `height` attributes.
```

File: src/site/content/en/blog/quintoandar/index.md [hunk 15]
```diff
@@ -0,0 +1,219 @@
...
+
+<figure class="w-figure">
+  {% Img src="image/jL3OLOhcWUQDnR4XjewLBx4e3PC3/QWkHnGHi4YTdBR6q17sw.png", alt="A line graph with FID values comparing the new and previous versions during the current and past month. The curve for the new version stays below 100ms most of the time, while in the curve for the previous version there are a few spikes crossing 250ms.
+", width="800", height="494" %}
+  <figcaption>
+    Field metrics results collected with Instana.
+  </figcaption>
+</figure>
+
+[PageSpeed Insights](https://pagespeed.web.dev/) provides a field data report for the last 28 days. [The most accessed condominium page](https://www.quintoandar.com.br/condominio/ed-copan-centro-historico-de-sao-paulo-sao-paulo-ndv7sq7j2d) alone had enough data to generate a report for mobile users. As of November 2021, all Core Web Vitals are in the “good” bucket.
```
Comment (heyawhite) [inline src/site/content/en/blog/quintoandar/index.md:186]:
```suggestion
[PageSpeed Insights](https://pagespeed.web.dev/) provides a field data report for the last 28 days. [The most accessed condominium page](https://www.quintoandar.com.br/condominio/ed-copan-centro-historico-de-sao-paulo-sao-paulo-ndv7sq7j2d) alone had enough data to generate a report for mobile users. As of November 2021, all Core Web Vitals are in the "good" bucket.
```

**Review by heyawhite — APPROVED** [review]
LGTM, with some small copy suggestions around em-dashes and quotes.

File: src/site/content/en/blog/quintoandar/index.md [hunk 16]
```diff
@@ -0,0 +1,219 @@
+---
+title: How QuintoAndar increased conversion rates and pages per session by improving page performance
+subhead: A project focused on optimizing Core Web Vitals and migrating to Next.js resulted in a 5% increase in conversion rates and a 87% increase in pages per session.
+authors:
+  - danielasy
+date: 2021-12-10
+hero: image/jL3OLOhcWUQDnR4XjewLBx4e3PC3/VQrzs5OoWZT2qS8yTzGW.jpg
+alt: A picture of a city skyline on a clear, sunny day. At left is a blue box with the QuintoAndar logo set within it in white.
```
Comment (danielasy) [inline src/site/content/en/blog/quintoandar/index.md:8]: Updating the hero alt text to match the new image:

```suggestion
alt: A picture of a city skyline on a clear, sunny day. At center is the QuintoAndar logo in white.
```

File: src/site/_data/authorsData.json [hunk 17]
```diff
@@ -1131,5 +1131,11 @@
     "country": "US",
     "twitter": "stubbornella",
     "image": "image/kheDArv5csY6rvQUJDbWRscckLr1/gjyoj6pRzTfjLXfU7589.jpeg"
+  },
+  "danielasy": {
+    "country": "BR",
+    "github": "danielasy",
+    "linkedin": "danielayassuda",
```
Comment (danielasy) [inline src/site/_data/authorsData.json:1138]: The link did not appear in the preview, so I assume this is the correct format:

```suggestion
    "linkedin": "https://www.linkedin.com/in/danielayassuda/",
```

**Review by danielasy — APPROVED** [review]
LGTM! Only two small adjustments

File: src/site/_data/authorsData.json (hunk 17 above)
Comment (malchata) [reply src/site/_data/authorsData.json:1138]: I haven't been able to get these to appear in posts. We can try that out and see if it works.

Comment (danielasy) [reply src/site/_data/authorsData.json]: Seems it didn't work
But I don't mind it that much, so don't consider this a blocker for merging

**Pull request merged**